.github/workflows/build.yml

```
@@ -0,0 +1,137 @@
+name: Build / Test / Push
+
+on:
+  push:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      build-image: ${{ steps.build-meta.outputs.tags }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Produce the build image tag
+        id: build-meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ github.repository }}
+          tags: type=sha,suffix=-build-${{ github.run_id }}_${{ github.run_attempt }}
+
+      - name: Build and push the untested image
+        uses: docker/build-push-action@v6
+        with:
+          push: true
+          labels: ${{ steps.build-meta.outputs.labels }}
+          tags: ${{ steps.build-meta.outputs.tags }}
+          provenance: true
+          sbom: true
+          target: development
+          cache-from: type=gha
+          cache-to: type=gha
+
+  test:
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        test:
+          - mypy .
+          - pydoclint .
+          # pylint returns error codes if the checks fail
+          # https://pylint.readthedocs.io/en/latest/user_guide/usage/run.html#exit-codes
+          - pylint -v .
+          - unittest -v
```

`unittest` is a built-in Python module, so it doesn't have its own script / binary:

```suggestion
          - python3 -m unittest -v
```